File: leakcanary-support-fragment/src/androidTest/java/com/squareup/leakcanary/tests/LeakingFragment.java
package com.squareup.leakcanary.tests;

import androidx.fragment.app.Fragment;

import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;

public class LeakingFragment extends Fragment {

  public static void add(final TestActivity activity) {
    getInstrumentation().runOnMainSync(new Runnable() {
      @Override public void run() {
        leakingFragment = new LeakingFragment();
        activity.getSupportFragmentManager()
            .beginTransaction()
            .add(0, leakingFragment)
            .commitNow();
      }
    });
  }

  private static LeakingFragment leakingFragment;
}

File: leakcanary-android-instrumentation/src/androidTest/java/leakcanary/LeakingFragment.kt
package leakcanary

import androidx.fragment.app.Fragment

import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation

class LeakingFragment : Fragment() {
  companion object {
    fun add(activity: TestActivity) {
      getInstrumentation().runOnMainSync {
        leakingFragment = LeakingFragment()
        activity.supportFragmentManager
            .beginTransaction()
            .add(0, leakingFragment)
            .commitNow()
      }
    }

    private lateinit var leakingFragment: LeakingFragment
  }
}
